src/Flysystem/FlysystemAssetStore.php

```diff
@@ -524,9 +524,12 @@ protected function isGranted($fileID)
             return true;
         }
         if ($member = Security::getCurrentUser()) {
-            $file = File::get()->byID($fileID);
```

Comment:
This will never return a file because $fileID will be a string like "foldername/xxxxxxxxxxx/filename.txt". `byID` expect a numeric ID.

PostgreSQL throws an exception here, because it doesn't like that we're trying to do a where statement against a numeric column with a string value. But this is equally problematic for a MySQL site.